### PR TITLE
Pre-commit: reintroduce pylint rules

### PR DESCRIPTION
**Pull request by ltalirz**
In 65ad067b18cffeb639994efe9a372ec1475e1615 the following pylint rules
were accidentally disabled:

 * missing-class-docstring
 * missing-function-docstring
 * too-many-ancestors
 * too-many-locals

This commit reintroduces all but the "too-many-ancestors" rule, which
was deemed not to be helpful, since it has never prompted refactoring in
the past.

### CODE REVIEW

**Comment by ltalirz** [comment]
@sphuber I started reducing the number of local variables in a few places until I realized that we had increased the acceptable limit to 20 ;-)

File: aiida/cmdline/commands/cmd_data/cmd_show.py [hunk 1]
```diff
@@ -10,6 +10,9 @@
 """
 This allows to manage showfunctionality to all data types.
 """
+import sys
```
Comment (greschd) [inline aiida/cmdline/commands/cmd_data/cmd_show.py:13]: Could those have been put in the functions to improve startup performance?

Comment (ltalirz) [reply aiida/cmdline/commands/cmd_data/cmd_show.py]: done!

you're right, here it makes sense to move imports inside functions (although those system libraries are likely imported somewhere anyhow)

**Review by sphuber — APPROVED** [review]
Thanks @ltalirz

**Pull request merged**